File: app/src/main/java/piuk/blockchain/android/ui/account/AccountEditView.java
package piuk.blockchain.android.ui.account;

import android.content.Intent;
import android.graphics.Bitmap;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;

import piuk.blockchain.android.ui.base.View;
import piuk.blockchain.android.ui.customviews.ToastCustom;

interface AccountEditView extends View {

    Intent getIntent();

    void promptAccountLabel(@Nullable String currentLabel);

    void showToast(@StringRes int message, @ToastCustom.ToastType String type);

    void setActivityResult(int resultCode);

    void startScanActivity();

    void promptPrivateKey(String message);

    void promptArchive(String title, String message);

    void promptBIP38Password(String data);

    void privateKeyImportMismatch();

    void privateKeyImportSuccess();

    void showXpubSharingWarning();

    void showAddressDetails(String heading, String note, String copy, Bitmap bitmap, String qrString);

    void showPaymentDetails(PaymentConfirmationDetails details);

    void showTransactionSuccess();

    void showProgressDialog(@StringRes int message);

    void dismissProgressDialog();

    void sendBroadcast(String key, String data);

    void updateAppShortcuts();

}

File: app/src/main/java/piuk/blockchain/android/ui/account/AccountEditView.kt
package piuk.blockchain.android.ui.account

import android.content.Intent
import android.graphics.Bitmap
import android.support.annotation.StringRes

import piuk.blockchain.android.ui.base.View
import piuk.blockchain.android.ui.customviews.ToastCustom

interface AccountEditView : View {

    val activityIntent: Intent

    fun promptAccountLabel(currentLabel: String?)

    fun showToast(@StringRes message: Int, @ToastCustom.ToastType type: String)

    fun setActivityResult(resultCode: Int)

    fun startScanActivity()

    fun promptPrivateKey(message: String)

    fun promptArchive(title: String, message: String)

    fun promptBIP38Password(data: String)

    fun privateKeyImportMismatch()

    fun privateKeyImportSuccess()

    fun showXpubSharingWarning()

    fun showAddressDetails(
            heading: String?,
            note: String?,
            copy: String?,
            bitmap: Bitmap?,
            qrString: String?
    )

    fun showPaymentDetails(details: PaymentConfirmationDetails)

    fun showTransactionSuccess()

    fun showProgressDialog(@StringRes message: Int)

    fun dismissProgressDialog()

    fun sendBroadcast(key: String, data: String)

    fun updateAppShortcuts()

    fun hideMerchantCopy()

    fun finishPage()

}
